_benchmark/configuring-benchmark.md

```diff
@@ -0,0 +1,202 @@
+| `async.debug` | Boolean | Enables debug mode on OpenSearch Benchmark's asyncio event loop. Default is `false`. |
+| `passenv` | String | A comma-separated list of environment variable names that should be passed to OpenSearch for processing. |
+
+## node
+
+This section contains node-specific information that can be customized according to the needs of your cluster.
+
+| Parameter | Type | Description |
+| :---- | :---- | :---- |
+| `root.dir` | String | The directory that stores all OpenSearch Benchmark data. OpenSearch Benchmark assumes control over this directory and all subdirectories therein. |
+| `src.root.dir` | String | The directory where the OpenSearch source code and any OpenSearch plugins are called. Only relevant for benchmarks from [sources](#source). |
+
+## source
+
+This section contains more details about the OpenSearch source tree.
+
+| Parameter | Type | Description |
+| :---- | :---- | :---- |
+| `remote.repo.url` | URL | The URL from which to checkout OpenSearch.  Default is `https://github.com/opensearch-project/OpenSearch.git`.
+| `opensearch.src.subdir` | String | The local path relative to `src.root.dir` of the OpenSearch search tree. Default is `OpenSearch`. 
+| `cache` | Boolean | Enables OpenSearch's internal source artifact cache `opensearch*.tar.gz` and any plugin zip files. Artifacts are cached based on their git revision. Default is `true`. |
+| `cache.days` | Integer | The number of days for which an artifact should be kept in the source artifact cache. Default is `7`. |
+
+## benchmarks
+
+This section contains the settings that can be customized in the OpenSearch Benchmark data directory.
+
+| Parameter | Type | Description |
+| :---- | :---- | :---- |
+| `local.dataset.cache` | String | The directory in which benchmark data sets are stored. Depending on the benchmarks that are run, this directory may contain hundreds of GB of data. Default path is `$HOME/.benchmark/benchmarks/data`. |
+
+## results_publishing
+
+This section defines how benchmark metrics are stored.
+
+| Parameter | Type | Description |
+| :---- | :---- | :---- |
+| `datastore.type` | String | If set to `in-memory` all metrics will be kept in memory while running the benchmark. If set to `opensearch` all metrics will instead be written to a persistent metrics store and the data
+are available for further analysis. Default is `in-memory`. |
+| `sample.queue.size` | Function | The number of metrics samples that can be stored in OpenSearch Benchmark’s in-memory queue. Default is `2^20`. | 
+| metrics.request.downsample.factor | Integer| (default: 1): Determines how many service time and latency samples should be kept in the metrics store. By default all values will be kept. If you want to, for example. keep only every 100th sample, specify `100`. This is useful to avoid overwhelming the metrics store in benchmarks with many clients. Default is `1`. |
```

Comment:
```suggestion
| metrics.request.downsample.factor | Integer| (default: 1): Determines how many service time and latency samples are saved in the metrics store. By default, all values are saved. If you want to, for example. keep only every 100th sample, specify `100`. This is useful to avoid overwhelming the metrics store in benchmarks with many clients. Default is `1`. |
```